### PR TITLE
L2 626 extend debug store

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

- `confirm` was replaced with `prompt` so it's not possible to accidentally triggers the debug output
- to be able to get original data new output types were added.

## prompt commands
 * `c` - pseudo-anonymised stringified -> console
 * `co` - original stringified -> console
 * `coo` - original as object -> console
 * `f` - pseudo-anonymised -> json file
 * `fo` - original -> json file

# Changes

- add ArrayBuffer support to stringifyJson (important for local mocking)
- confirm to prompt
- possibility to output not pseudo-anonymised data
- add more data to debug outputs

# Tests

https://user-images.githubusercontent.com/98811342/171833622-c9dbe3fa-01bf-4853-b178-621aa637c51a.mov

### CODE REVIEW

File: frontend/svelte/src/lib/utils/anonymize.utils.ts [hunk 1]
```diff
@@ -111,7 +93,7 @@ export const anonymizeAccount = async (
 
   return {
     identifier: await cutAndAnonymize(identifier),
-    principal: isNullOrUndefined(principal) ? "yes" : "no",
+    principal: isNullOrUndefined(principal) ? "no" : "yes",
```
Comment (peterpeterparker) [inline frontend/svelte/src/lib/utils/anonymize.utils.ts:96]: that was a mistake in previous version or that's a mistake in new version?

Comment (mstrasinskis) [reply frontend/svelte/src/lib/utils/anonymize.utils.ts:96]: It was a mistake.

Comment (peterpeterparker) [reply frontend/svelte/src/lib/utils/anonymize.utils.ts:96]: 🤪

File: frontend/svelte/src/lib/services/debug.services.ts [hunk 2]
```diff
@@ -94,18 +109,36 @@ export const generateDebugLog = async (safeToFile: boolean) => {
...
+};
 
+/**
+ * 1. generates anonymized version of stores state
+ * 2. log it in the dev console
+ * 3. generates a json file with logged context
+ */
+export const generateDebugLog = async (logType: LogType) => {
+  const debugStore = initDebugStore();
+  const anonymise = logType === LogType.Console || logType === LogType.File;
```
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/svelte/src/lib/services/debug.services.ts]: Looks better 👍

File: frontend/svelte/src/lib/utils/anonymize.utils.ts [hunk 3]
```diff
@@ -230,7 +212,7 @@ export const anonymizeCanister = async (
     name,
     // TODO: what to do with principals
     // canister_id: await anonymize(canister_id),
-    canister_id: isNullOrUndefined(canister_id) ? "yes" : "no",
+    canister_id: isNullOrUndefined(canister_id) ? "no" : "yes",
```
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/svelte/src/lib/utils/anonymize.utils.ts]: Good point, done with `anonymiseAvailability`

**Pull request merged**